src/librustc_mir/borrow_check/nll/type_check/mod.rs

```diff
@@ -2210,38 +2210,39 @@ impl<'a, 'tcx> TypeChecker<'a, 'tcx> {
                                 }
                             }
 
-                            if let hir::MutMutable = mutability {
```

Comment:
why did you change this to a match? A match will increase the nesting here, and for mutability it seems like there's not too much to be gained from having an exhaustive match. We're not going to extend that enum in the future with more mutability variants I believe

Reply:
I tend to think exhaustive matches as less demanding in terms of brain bandwidth. However in that case, I agree and reverted the change.